arches_search/utils/advanced_search/related_clause_evaluator.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Optional
+from django.db.models import Exists, OuterRef, Q
```

Comment:
Consider if my suggestions below make sense.
```suggestion
from django.db.models import Exists, OuterRef, Q
from django.utils.translation import gettext as _
```